### PR TITLE
 Support serviceAccountEmail and GS Variable

### PR DESCRIPTION
**Pull request by luisgreen**
## What did you implement
Added support for Google Cloud Storage variable fetch and Service Account Email 

## How can we verify it

`serverless.yaml`

```yaml
service: my-super-functions

provider:
  name: google
  stage: dev
  runtime: nodejs8
  region: us-central1
  project: my-super-project

plugins:
  - serverless-google-cloudfunctions

package:
  exclude:
    - node_modules/**
    - .gitignore
    - .git/**

functions:
  gcp-sites:
    #<FUNCTION DEFINITION GOES HERE>
    serviceAccountEmail: devops-functions@${self:provider.project}.iam.gserviceaccount.com
    environment:
      ALGO: ${gs:my-superbucket/myfileinbucket.txt}
```

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: provider/googleProvider.js [hunk 1]
```diff
@@ -41,6 +42,29 @@ class GoogleProvider {
...
+    const object = groups.join('/');
+
+    return this.serverless
+      .getProvider('google')
+      .request('storage', 'objects', 'get', {
+        bucket,
+        object,
+        alt: 'media',
+      })
+      .then(response => BbPromise.resolve(response))
```
Comment (medikoo) [inline provider/googleProvider.js:63]: This line is obsolete (has no effect)

File: provider/googleProvider.js [hunk 2]
```diff
@@ -41,6 +42,29 @@ class GoogleProvider {
...
+      .catch(err => {
+        const errorMessage = `Error getting value for ${variableString}. ${err.message}`;
+        return BbPromise.reject(new this.serverless.classes.Error(errorMessage));
```
Comment (medikoo) [inline provider/googleProvider.js:66]: Let's just throw error, it's more natural, and effect will be same

File: package/lib/compileFunctions.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ module.exports = {
         `gs://${this.serverless.service.provider.deploymentBucketName}/${this.serverless.service.package.artifactFilePath}`
       );
 
+      funcTemplate.properties.serviceAccountEmail =
+        _.get(funcObject, 'serviceAccountEmail') ||
+        _.get(this, 'serverless.service.provider.serviceAccountEmail') ||
+        '';
```
Comment (medikoo) [inline package/lib/compileFunctions.js:38]: `null` as fallback seems more natural

File: package/lib/compileFunctions.js [hunk 4]
```diff
@@ -48,6 +52,10 @@ module.exports = {
         funcObject.environment // eslint-disable-line comma-dangle
       );
 
+      if (!funcTemplate.properties.serviceAccountEmail) {
+        delete funcTemplate.properties.serviceAccountEmail;
+      }
```
Comment (medikoo) [inline package/lib/compileFunctions.js:57]: Why is that needed?

Comment (luisgreen) [reply package/lib/compileFunctions.js:57]: Because GCP cannot receive the variable as empty, (throws error) we need ro remove it from the whole structure.

File: package/lib/compileFunctions.js (hunk 3 above)
Comment (luisgreen) [reply package/lib/compileFunctions.js:38]: On my way.

File: provider/googleProvider.js (hunk 1 above)
Comment (luisgreen) [reply provider/googleProvider.js:63]: On my way!

File: provider/googleProvider.js (hunk 2 above)
Comment (luisgreen) [reply provider/googleProvider.js:66]: On my way!

File: package/lib/compileFunctions.js (hunk 4 above)
Comment (luisgreen) [reply package/lib/compileFunctions.js:57]: But you are right... the template does not have it at this stage, so, is not needed actually...

Comment (luisgreen) [reply package/lib/compileFunctions.js]: Sorry, tested thoroughly and this piece of code needs to be in place when null:
 `       
error: instance type (null) does not match any allowed primitive type (allowed: ["string"])
      level: "error"
      schema: {"loadingURI":"#","pointer":"/create/properties/serviceAccountEmail"}
      instance: {"pointer":"/serviceAccountEmail"}
      domain: "validation"
      keyword: "type"
      found: "null"
      expected: ["string"]
`

Comment (medikoo) [reply package/lib/compileFunctions.js:57]: And what would be the case if we'd fallback to `undefined` instead of `null` ?

Comment (luisgreen) [reply package/lib/compileFunctions.js:57]: Falled back to undefined also fails:

<details>
<summary>Click here to see Api Error from GCP</summary>

>   Error: Error parsing configuration: could not determine a constructor for the tag tag:yaml.org,2002:js/undefined
>    in 'string', line 52, column 28:
>             serviceAccountEmail: !<tag:yaml.org,2002:js/undefined> ''
>                                  ^
> 
>       at Gaxios._request (D:\CLOUD_SANDBOX\EVALUESERVE\CODE\CloudFunctions-NodeJS\node_modules\gaxios\src\gaxios.ts:109:15)
>       at process._tickCallback (internal/process/next_tick.js:68:7)

</details>

Comment (medikoo) [reply package/lib/compileFunctions.js:57]: Ok, thanks for investigating that
